app/Api/Controllers/NotificationController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Api\Controllers;
+
+use Dingo\Api\Http;
+use Dingo\Api\Routing\Helpers;
+use App\Notification;
+use App\NotificationAttrib;
+use Illuminate\Http\Request;
+
+class NotificationController extends Controller
+{
+
+    use Helpers;
+
+    public function __construct() {
+    }
+
+    /**
+     * Display a listing of all notifications
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request, $type = null)
+    {
+        if ($type === 'archive')
+        {
+            $notifications = Notification::read($request)->get();
+        }
+        else {
+            $notifications = Notification::unread()->get();
+        }
+        return $notifications;
+    }
+
+    public function update(Request $request, $id, $action)
+    {
+        if ($action === 'read' || $action === 'sticky')
+        {
+            if (NotificationAttrib::where('notifications_id', $id)->delete() >= 0)
```

Comment:
If someone can show me working code of what you mean that would be good. I managed to break it trying to do what I think you mean so reverted it back to as is now.